### PR TITLE
Flags for parameters as requested in issue #8

### PR DESCRIPTION
**Pull request by PifPof73**
Hi,
as discussed yesterday here is my first pull request as suggestion for fullfilling the feature request on putting flags to parameters like a direction for example.
The flag signs are at the moment [], but I followed the logic of the existing code to keep that easy expandable.
There is no extra formatting in the output at the moment.
Regards,
PifPof73

### CODE REVIEW

File: source/delphi_gui/pasdoc_gui.dpr [hunk 1]
```diff
@@ -1,4 +1,4 @@
-program pasdoc_gui;
+program pasdoc_gui_Rio;
```
Comment (michaliskambi) [inline source/delphi_gui/pasdoc_gui.dpr:1]: Same comment here: do not rename programs that are not only for Rio :) Please revert this change, so program name stays `pasdoc_gui`.

File: source/delphi_gui/pasdoc_gui_Tokyo.dproj [hunk 2]
```diff
@@ -1,206 +0,0 @@
-﻿<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (michaliskambi) [inline source/delphi_gui/pasdoc_gui_Tokyo.dproj:1]: Same comment as before: do not remove Delphi Tokyo specific files.

File: source/tools/file_to_pascal_data.dpr [hunk 3]
```diff
@@ -31,7 +31,7 @@
   Useful if you want to "embed" some binary file inside compiled
   Pascal program. }
 
-program file_to_pascal_data;
+program file_to_pascal_data_Rio;
```
Comment (michaliskambi) [inline source/tools/file_to_pascal_data.dpr:34]: Same as above: keep original program name, without _Rio suffix.

File: source/tools/file_to_pascal_string.dpr [hunk 4]
```diff
@@ -30,7 +30,7 @@
   Note it treats $1 as text file, so difference between various line endings
   is lost in $2 (line endings are always encoded using LineEnding constant). }
 
-program file_to_pascal_string;
+program file_to_pascal_string_Rio;
```
Comment (michaliskambi) [inline source/tools/file_to_pascal_string.dpr:33]: Same as above: keep original program name, without _Rio suffix.

File: source/tools/pascal_pre_proc.dpr [hunk 5]
```diff
@@ -33,7 +33,7 @@
   Things like $mode, or $H+.
 }
 
-program pascal_pre_proc;
+program pascal_pre_proc_Rio;
```
Comment (michaliskambi) [inline source/tools/pascal_pre_proc.dpr:36]: Same as above: keep original program name, without _Rio suffix.

File: source/tools/file_to_pascal_data_Tokyo.dproj [hunk 6]
```diff
@@ -1,220 +0,0 @@
-﻿<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (michaliskambi) [inline source/tools/file_to_pascal_data_Tokyo.dproj:1]: Same as above: do not delete Delphi Tokyo files.

File: source/tools/file_to_pascal_string_Tokyo.dproj [hunk 7]
```diff
@@ -1,220 +0,0 @@
-﻿<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (michaliskambi) [inline source/tools/file_to_pascal_string_Tokyo.dproj:1]: Same as above: do not delete Delphi Tokyo files.

File: source/tools/pascal_pre_proc_Tokyo.dproj [hunk 8]
```diff
@@ -1,222 +0,0 @@
-﻿<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (michaliskambi) [inline source/tools/pascal_pre_proc_Tokyo.dproj:1]: Same as above: do not delete Delphi Tokyo files.

File: source/component/PasDoc_Types.pas [hunk 9]
```diff
@@ -196,7 +196,7 @@ function SplitNameParts(S: string;
     Inc(i);
   end;
 
-  NameParts := SplitString(s, '.');
+  NameParts := TNameParts(SplitString(s, '.'));
```
[the remark this answers is not in the document]
Comment (PifPof73) [reply source/component/PasDoc_Types.pas:199]: Hi, 
the source was not compilable under Delphi Reio, therefor I just made the typecast as I do not wanted to change your code completly without knowing possible site effects.
You can leave it - than the code does not compile in Delphi, or I change it as you suggested.
Regards,
PifPof73

File: source/console/pasdoc.dpr [hunk 10]
```diff
@@ -48,7 +48,7 @@
 
 ------------------------------------------------------------------------------ }
 
-program pasdoc;
+program pasdoc_Rio;
```
[the remark this answers is not in the document]
Comment (PifPof73) [reply source/console/pasdoc.dpr:51]: ok, sorry for that.

File: source/console/pasdoc_Tokyo.dproj [hunk 11]
```diff
@@ -1,141 +0,0 @@
-﻿<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
[the remark this answers is not in the document]
Comment (PifPof73) [reply source/console/pasdoc_Tokyo.dproj:1]: I am using Delphi Rio (10.3).
I didn't want to change your part and I don't know if you have specific Tokyo code.
Therefor I just generated Rio Projects.
But I do not do anything there.
So I can revert the changes - as you have specific "Tokyo" project I was misguided by reviewing your code.

File: source/delphi_gui/pasdoc_gui.dpr (hunk 1 above)
Comment (PifPof73) [reply source/delphi_gui/pasdoc_gui.dpr:1]: Same reason here.

File: source/delphi_gui/pasdoc_gui_Tokyo.dproj (hunk 2 above)
Comment (PifPof73) [reply source/delphi_gui/pasdoc_gui_Tokyo.dproj:1]: Same reason here.

File: source/tools/file_to_pascal_string.dpr (hunk 4 above)
Comment (PifPof73) [reply source/tools/file_to_pascal_string.dpr:33]: Same reason here.

File: source/tools/pascal_pre_proc.dpr (hunk 5 above)
Comment (PifPof73) [reply source/tools/pascal_pre_proc.dpr:36]: Same reason here.

File: source/tools/file_to_pascal_data_Tokyo.dproj (hunk 6 above)
Comment (PifPof73) [reply source/tools/file_to_pascal_data_Tokyo.dproj:1]: Same reason here.

File: source/tools/file_to_pascal_string_Tokyo.dproj (hunk 7 above)
Comment (PifPof73) [reply source/tools/file_to_pascal_string_Tokyo.dproj:1]: Same reason here.

File: source/tools/pascal_pre_proc_Tokyo.dproj (hunk 8 above)
Comment (PifPof73) [reply source/tools/pascal_pre_proc_Tokyo.dproj:1]: Same reason here.

File: source/component/PasDoc_Types.pas (hunk 9 above)
[the remark this answers is not in the document]
Comment (michaliskambi) [reply source/component/PasDoc_Types.pas:199]: I understand. I would ask to change as I suggested (make `TNameParts = TStringArray) -- this way it compiles with Rio, and looks clean (without extra typecast) :)

You can push more commits (to the same branch of your GIT fork of PasDoc) and then they will automatically be added to this pull request.

File: source/console/pasdoc_Tokyo.dproj (hunk 11 above)
[the remark this answers is not in the document]
Comment (michaliskambi) [reply source/console/pasdoc_Tokyo.dproj:1]: Yes, please revert these changes, so that project files for "Tokyo" stay in the repository.

You can push additional commits to GIT, and then they will be added to this pull request. Thank you!

**Comment by PifPof73** [comment]
I made a new fork and pull request to be faster in reverting my unnecessary changes.
